packages/publisher/gcs/src/Config.ts

```diff
@@ -0,0 +1,49 @@
+export interface PublisherGCSConfig {
+  /**
+   * The path to the file that is either:
+   * - the JSON file that contains your Google service account credentials, or
+   * - the PEM/PKCS #12-formatted file that contains the private key.
+   *
+   * Defaults to the value in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+   */
+  keyFilename?: string;
+  /**
+   * The Google Cloud project ID.
+   *
+   * Defaults to the value in the `GOOGLE_CLOUD_PROJECT` environment variable.
+   * */
+  projectId?: string;
+  /**
+   * The email for your Google service account, *required* when using a PEM/PKCS #12-formatted
+   * file in the [[keyFilename]] option.
+   *
+   * Defaults to the value in the `GOOGLE_CLOUD_CLIENT_EMAIL` environment variable.
+   */
+  clientEmail?: string;
+  /**
+   * The private key for your Google service account.
+   *
+   * Defaults to the value in the `GOOGLE_CLOUD_PRIVATE_KEY` environment variable.
+   */
+  privateKey?: string;
```

Comment:
comment: For these auth-related options, can we put them in a separate `auth` object?

Reply:
I think this is still a valid concern. It might be cleaner to put all auth-related private key config options under an `auth` property.